clis/teliod/src/config.rs

```diff
@@ -50,11 +53,47 @@ fn reconnect_after_expiry_default() -> Percentage {
     Percentage(90)
 }
 
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct DeviceIdentity {
+    pub hw_identifier: String,
+    pub private_key: SecretKey,
+    pub machine_identifier: Option<String>,
+}
+
+impl DeviceIdentity {
+    pub async fn new(identity_path: &PathBuf) -> DeviceIdentity {
+        info!("Fetching config");
+
+        if let Ok(file) = fs::File::open(identity_path) {
+            debug!("found existing config.");
+            if let Ok(c) = serde_json::from_reader(file) {
+                return c;
+            }
+        }
+
+        // Generate Secret/public key pair
+        debug!("Generating secret key!");
+        let private_key = SecretKey::gen();
+        let public_key = private_key.public();
+
+        // Generate hw_identifier
+        debug!("Generating hw identifier");
+        let hw_identifier = format!("{}.{}", public_key, "teliod");
+
+        DeviceIdentity {
+            hw_identifier,
+            private_key,
+            machine_identifier: None,
```

Comment:
I think it shouldn't be an option - a ready `DeviceIdentity` should have all of the fields ready - soo, I think you can just have two constructors? `from_path` and `new(machine_id)` (or maybe `generate(machine_id)`?). If you fail to get data from file, you can then fetch identifier from File and generate a new one.

Reply:
Added `from_path` constructor